src/Middleware/JsonSchema.php

```diff
@@ -0,0 +1,98 @@
+<?php
+
+namespace Psr7Middlewares\Middleware;
+
+use JsonSchema\Validator;
+use Psr\Http\Message\ResponseInterface;
+use Psr\Http\Message\ServerRequestInterface;
+
+class JsonSchema
+{
+    /** @var string[] */
+    private $schemas;
+
+    /**
+     * JsonSchema constructor.
+     * @param string[] $schemas [uri => file]
+     */
+    public function __construct(array $schemas)
+    {
+        $this->schemas = $schemas;
+    }
+
+    /**
+     * Execute the middleware.
+     *
+     * @param ServerRequestInterface $request
+     * @param ResponseInterface      $response
+     * @param callable               $next
+     *
+     * @return ResponseInterface
+     */
+    public function __invoke(ServerRequestInterface $request, ResponseInterface $response, callable $next)
+    {
+        if ($schema = $this->getSchema($request)) {
+            $validator = new Validator();
+            $decoded = $this->decodeBody($request);
+            if (json_last_error() !== JSON_ERROR_NONE) {
+                return $response->withStatus(422, 'Malformed JSON: ' . json_last_error_msg());
+            }
+
+            $validator->check($decoded, (object) [
+                '$ref' => $this->normalizeSchema($schema),
+            ]);
+
+            if (!$validator->isValid()) {
+                $response->getBody()->write(stripslashes(json_encode($validator->getErrors())));
```

Comment:
Please, use the constants options like `json_encode($validator->getErrors(), JSON_UNESCAPED_SLASHES)`